File: code/python/delete1.py
#!/usr/bin/env python

import sys
import xapian

### Start of example code.
def delete_docs(dbpath, identifiers):
    # Open the database we're going to be deleting from.
    db = xapian.WritableDatabase(dbpath, xapian.DB_OPEN)

    for identifier in identifiers:
        idterm = u'Q' + identifier
        db.delete_document(idterm)
### End of example code.

if len(sys.argv) < 3:
    print("Usage: %s DBPATH ID..." % sys.argv[0])
    sys.exit(1)

delete_docs(dbpath = sys.argv[1], identifiers=sys.argv[2:])

File: code/c++/delete1.cc
#include <xapian.h>

#include <iostream>
#include <string>

using namespace std;

// Start of example code.
static void delete_docs(const string &dbpath, char ** identifiers)
{
    // Open the database we're going to be deleting from.
    Xapian::WritableDatabase db(dbpath, Xapian::DB_OPEN);

    while (*identifiers) {
	string idterm = "Q";
	idterm += *identifiers;
	db.delete_document(idterm);
	++identifiers;
    }
}
// End of example code.

int main(int argc, char** argv) {
    if (argc < 3) {
	cerr << "Usage: " << argv[0] << " DBPATH ID...\n";
	return 1;
    }
    delete_docs(argv[1], argv + 2);
}
